File: keeping up with the javascripts part 1 es6/Homeworks/06 - Loops/fizzbuzz.js
/*
 * Homework #6: Loops
 * Keeping Up With the Javascripts - Part 1: ES6
 * Pirple
 * 
 * Write a program that prints the numbers from 1 to 100. But for multiples of
 * three print "Fizz" instead of the number and for the multiples of five print "Buzz".
 * For numbers which are multiples of both three and five print "FizzBuzz".
 * 
 * Alejandro Mohamad
 */

const isPrime = (number) => {
  let start = 2;
  const limit = Math.sqrt(number);
  while (start <= limit) {
      if (number % start++ < 1) return false;
  }
  return number > 1;
}

for (let position = 1; position <= 100; position++) {
  if (isPrime(position)) {
    console.log("Prime");
  } else if (position % 3 === 0 && position % 5 === 0) {
    console.log("FizzBuzz");
  } else if (position % 3 === 0) {
    console.log("Fizz");
  } else if (position % 5 === 0) {
    console.log("Buzz");
  } else {
    console.log(position);
  }
}

File: get swifty part 1/Homeworks/03 - Control Flow/fizzbuzz.swift
/*
 *  Homework #3: Control Flow
 *  Get Swifty - Part 1
 *  Pirple
 *
 *  Write a program that prints the numbers from 1 to 100. But for
 *  multiples of three print "Fizz" instead of the number and for the
 *  multiples of five print "Buzz". For numbers which are multiples of
 *  both three and five print "FizzBuzz".
 *
 *  Alejandro Mohamad
 */

func prime(_ number: Int) -> Bool {
    return (1...number).filter({number % $0 == 0}).count <= 2
}

for number in 1...100 {
    if prime(number) {
        print("Prime")
    } else if number % 3 == 0 && number % 5 == 0 {
        print("FizzBuzz")
    } else if number % 3 == 0 {
        print("Fizz")
    } else if number % 5 == 0 {
        print("Buzz")
    } else {
        print(number)
    }
}
